mantidimaging/gui/windows/spectrum_viewer/test/presenter_test.py

```diff
@@ -285,18 +289,34 @@ def test_WHEN_rits_roi_clicked_THEN_rois_not_updated(self):
     def test_WHEN_ROI_renamed_THEN_roi_renamed(self):
         rois = ["all", "roi", "roi_1"]
         self.view.spectrum_widget.roi_dict = {roi: mock.Mock() for roi in rois}
-        self.presenter.model._roi_ranges = {roi: mock.Mock() for roi in rois}
-        self.view.spectrum_widget.rename_roi = mock.Mock()
+        self.view.spectrum_widget.rois = {roi: mock.Mock() for roi in rois}
+        self.view.spectrum_widget.rename_roi = mock.Mock(
+            side_effect=lambda old_name, new_name: self.view.spectrum_widget.rois.update(
+                {new_name: self.view.spectrum_widget.rois.pop(old_name)}))
+
         self.presenter.rename_roi("roi_1", "new_name")
         self.view.spectrum_widget.rename_roi.assert_called_once_with("roi_1", "new_name")
-        self.assertIn("new_name", self.presenter.model._roi_ranges)
-        self.assertNotIn("roi_1", self.presenter.model._roi_ranges)
+        self.assertIn("new_name", self.view.spectrum_widget.rois)
+        self.assertNotIn("roi_1", self.view.spectrum_widget.rois)
```

Comment:
Using the lambda means this is only really testing the test. As this is a test for the presenter, then only bit to test is that it calls the function in spectrum widget.